jetty-core/jetty-compression/jetty-compression-api/src/main/java/org/eclipse/jetty/compression/Compression.java

```diff
@@ -0,0 +1,355 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.compression;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+import org.eclipse.jetty.http.EtagUtils;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.io.ByteBufferPool;
+import org.eclipse.jetty.io.Content;
+import org.eclipse.jetty.io.RetainableByteBuffer;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.component.Container;
+import org.eclipse.jetty.util.component.ContainerLifeCycle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class Compression extends ContainerLifeCycle
+{
+    private static final Logger LOG = LoggerFactory.getLogger(Compression.class);
+    private final String encodingName;
+    private final String etagSuffix;
+    private final String etagSuffixQuote;
+    private ByteBufferPool byteBufferPool;
```

Comment:
I can only talk about the decision to use this vs ByteBufferPool.Sized at the moment.
I'll change it to ByteBufferPool.Sized now, and see what happens.
The history: The ByteBufferPool.Sized was quite buggy when I wrote this.  It didn't behave properly at that point in time.  
Also, at the time, (no longer true), the raw ByteBuffer was going through internals of the compression libs and using the RetainableByteBuffer concepts it made it difficult to track the specific buffer that was held in the compression lib to the buffer from this pool.

Reply:
If a ByteBufferPool is unconfigured here, then the default ByteBufferPool.NON_POOLING is used, and since not all ByteBufferPool implementations are ByteBufferPool.Sized, this was the best and most flexible solution.   If the implementation wants to use ByteBufferPool.Sized, then a call to .setByteBufferPool(new ByteBufferPool.Sized()) will still work here.